### PR TITLE
[EASI-4159] Graphql codegen translation schema

### PR DESCRIPTION
**Pull request by patrickseguraoddball**
# EASI-4159

## Changes and Description

- Extracted base translations field properties and added them to schema for type generation
- Added translation schemas for each task list section
- Mapped generated base type with extra FE types
- Merged generated task list section schema with FE form translation schema
- Added `TranslationDataType` in favor of string literal
- Added lint ignore for schemas that are not used by BE

## How to test this change

There should be no UI changes.  App should behave as normal

## PR Author Review Checklist

- [ ] Met the ticket's acceptance criteria, or will meet them in a subsequent PR.
- [ ] Added or updated tests for backend resolvers or other functions as needed.
- [ ] Added or updated client tests for new components, parent components, functions, or e2e tests as necessary.
- [ ] Updated the [Postman Collection](../MINT.postman_collection.json) if necessary.

## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.

### CODE REVIEW

File: pkg/graph/schema/types/translation.graphql [hunk 1]
```diff
@@ -0,0 +1,62 @@
+"""
+Base typea that represents FE translation structure
```
Comment (StevenWadeOddball) [inline pkg/graph/schema/types/translation.graphql:2]: Is this a block comment for the file?  that comment structure attaches as a description in the GQL Schema. 
(in this case to the TranslationDataType). Could you change the page comment to use `#` at the start of every line instead?

if you update the type comments to use `"""`, they will be exposed in the GQL introspection query

File: pkg/graph/schema/types/translation.graphql [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+enum TranslationDataType {
+  STRING
+  NUMBER
+  BOOLEAN
+  DATE
+  ENUM
+  OBJECT
+}
+
+# Represents the FORM type of the translation field
```
Comment (StevenWadeOddball) [inline pkg/graph/schema/types/translation.graphql:16]:
```suggest
` ` ` (replace without spaces)
 Represents the FORM type of the translation field
` ` `
```
image

File: pkg/graph/schema/types/translation.graphql [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  dbField: String!
+  label: String!
+  readonlyLabel: String
+  sublabel: String
+  multiSelectLabel: String
+  isArray: Boolean
+  dataType: TranslationDataType!
+  formType: TranslationFormType!
+  isOtherType: Boolean # Is a question a followup to another that doesn't designate it's own readonly question/line,
+  otherParentField: String # gql field name for the parent question for fields that represent Other, Please specify, etc.  Used in change history to render parent question for context
```
Comment (StevenWadeOddball) [inline pkg/graph/schema/types/translation.graphql:43]: Should this say the db_field name for the parent question? Or is that just dependent on the data export?

Also, if you use the ` ` ` comment style on the line before this field, it is exported in the introspection query

**Comment by StevenWadeOddball** [comment]
This looks great Patrick! I just added some notes for adding comments to the file, but doesn't change functionality. I think this will work great to run with!

File: pkg/graph/schema/types/translation.graphql (hunk 1 above)
Comment (patrickseguraoddball) [reply pkg/graph/schema/types/translation.graphql:2]: Gotcha, thanks I have no idea how comments work here.  I just copied from the other related file haha

File: pkg/graph/schema/types/translation.graphql (hunk 3 above)
Comment (patrickseguraoddball) [reply pkg/graph/schema/types/translation.graphql:43]: The export converts it to the db field.  In the frontend, it is a gqlfield

**Comment by patrickseguraoddball** [comment]
> This looks great Patrick! I just added some notes for adding comments to the file, but doesn't change functionality. I think this will work great to run with!

Thanks I'll update these comments :)

File: pkg/graph/schema/types/translation.graphql (hunk 3 above)
Comment (StevenWadeOddball) [reply pkg/graph/schema/types/translation.graphql:43]: That works! I'm going to manually implement the type on the backend, so I can specify it's a db field name on the go type

Comment (patrickseguraoddball) [reply pkg/graph/schema/types/translation.graphql:43]: I'm also open to changing it to the db field, as currently the FE isn't using this in any way yet.  If it's just the BE use we can maybe convert that data

Comment (StevenWadeOddball) [reply pkg/graph/schema/types/translation.graphql:43]: Thanks Patrick! I'm good with trying any way to start it out. We could also just take out the GQL specific language, and just leave it to say field name. That way, it's true for FE and BE use cases?

**Pull request merged**